patches/tModLoader/Terraria/ID/SetFactory.TML.cs

```diff
@@ -16,14 +18,23 @@ namespace Terraria.ID;
 public partial class SetFactory
 {
 	// Additional code to support named custom sets for ad-hoc collaboration
+
+	// Contains all SetFactory instances relating to named ID sets.
+	internal static HashSet<SetFactory> setFactories = new HashSet<SetFactory>();
+
 	private record SetMetadata(object defaultValue, object array);
+	private record SetNameTypePair(string setName, Type type);
 
-	private ConcurrentDictionary<(string, Type), SetMetadata> setMetadataMapping = new ConcurrentDictionary<(string, Type), SetMetadata>();
+	private ConcurrentDictionary<SetNameTypePair, SetMetadata> setMetadataMapping = new ConcurrentDictionary<SetNameTypePair, SetMetadata>();
+	private ConcurrentDictionary<SetNameTypePair, HashSet<string>> involvedMods = new ConcurrentDictionary<SetNameTypePair, HashSet<string>>();
+	private ConcurrentDictionary<SetNameTypePair, Dictionary<string, string>> setDescriptions = new ConcurrentDictionary<SetNameTypePair, Dictionary<string, string>>();
```

Comment:
Seems like these should be merged, with one class containing all the values here